src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/GeneralInvocationExpression.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using CS = Microsoft.CodeAnalysis.CSharp;
+using VB = Microsoft.CodeAnalysis.VisualBasic;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common
+{
+    public class GeneralInvocationExpression
+    {
+        private readonly SyntaxNode _invocationExpression;
+
+        public GeneralInvocationExpression(SyntaxNode syntaxNode)
+        {
+            if (!syntaxNode.IsInvocationExpression())
+            {
+                throw new ArgumentException("Invalid type of syntaxNode", nameof(syntaxNode));
+            }
+
+            // the only valid way to construct one is to try parse
+            // this enables validation to assert we're providing a wrapper around a valid SyntaxNode
+            _invocationExpression = syntaxNode ?? throw new ArgumentNullException(nameof(syntaxNode));
+        }
+
+        public IEnumerable<SyntaxNode> GetArguments()
```

Comment:
nit: IEnumerable<> ends up allocating a bit. Not sure where this is on a fast path, but it may be better to just return the SyntaxNodeOrList